File: 2024 czerwiec/INF.04-02-24.06-SG/Aplikacja Konsolowa/C++/main.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

class Album
{
public:
    string artist;
    string album;
    int songsNumber;
    int year;
    int downloadNumber;

    Album(string artist, string album, int songsNumber, int year, int downloadNumber) : artist(artist), album(album), songsNumber(songsNumber), year(year), downloadNumber(downloadNumber)
    {
    }

    /*
        * nazwa funkcji: display
        * opis funkcji: wyświetla dane albumu
        * parametry: brak
        * zwracany typ i opis: brak
        * autor: 00000000000
    */
    void display() const
    {
        cout << artist << endl
             << album << endl
             << songsNumber << endl
             << year << endl
             << downloadNumber << "\n\n";
    }
};

vector<Album> readAlbums() {
    ifstream albumsFile("Data.txt");

    string artist, album;
    int songsNumber, year, downloadNumber;

    int lineIndex = 0;
    string line;

    vector<Album> albums;
    while (getline(albumsFile, line))
    {
        lineIndex++;
        switch (lineIndex)
        {
        case 1:
            artist = line;
            break;
        case 2:
            album = line;
            break;
        case 3:
            songsNumber = stoi(line);
            break;
        case 4:
            year = stoi(line);
            break;
        case 5:
            downloadNumber = stoi(line);
            break;
        case 6:
            albums.emplace_back(artist, album, songsNumber, year, downloadNumber);
            lineIndex = 0;
            break;
        
        default:
            break;
        }
    }
    albumsFile.close();

    return albums;
    
}

int main()
{
    vector<Album> albums = readAlbums();

    for (const Album& album : albums) {
        album.display();
    }

    return 0;
}

File: 2024 czerwiec/INF.04-02-24.06-SG/Aplikacja Konsolowa/Python/main.py
class Album:
    def __init__(self, artist, album, songsNumber, year, downloadNumber):
        self.artist = artist
        self.album = album
        self.songsNumber = songsNumber
        self.year = year
        self.downloadNumber = downloadNumber
        
    """
        nazwa funkcji: display
        opis funkcji: wyświetla dane albumu
        parametry: brak
        zwracany typ i opis: brak
        autor: 00000000000
    """
    def display(self):
        print(self.artist, self.album, self.songsNumber, self.year, self.downloadNumber, sep="\n", end="\n\n")

def readAlbums():
    albums = []

    with open("Data.txt", "r", encoding="utf-8") as albumsFile:
        lineIndex = 0
        artist = ""
        album = ""
        songs_number = 0 
        year = 0
        download_number = 0
        
        for line in albumsFile:
            line = line.strip()

            lineIndex += 1

            if lineIndex == 1:
                artist = line
            elif lineIndex == 2:
                album = line
            elif lineIndex == 3:
                songs_number = int(line)
            elif lineIndex == 4:
                year = int(line)
            elif lineIndex == 5:
                download_number = int(line)
            elif lineIndex == 6:
                albums.append(Album(artist, album, songs_number, year, download_number))
                lineIndex = 0
    
    return albums

albums = readAlbums()

for album in albums:
    album.display()